Ud sdk update

bwockner opened the pull request:
Update to use udSDK
implemented udQueryFilter
WIP on udQueryContext and udRenderPicking

Commit:
removed deprecated license handling functions;
fixed issue with ortho camera that inversed zoom

Commit:
Fixed bug in flythrough introduced by changing position to a property
Extended API to support more enums and Streamer in preparation for udStream upgrade
Made resolution match display instead of being hard coded
Changed default server used to udStream

Commit:
Updated to work with udSDK on windows. Changed all function calls and added some that weren't present
Added udRenderSettings and picking to EasyRenderer: tested to correctly read picking values and set point mode

Commit:
Added custom function keymapping example
Added box query filter class

Commit:
Updated documentation

Commit:
Added attribute enums
WIP on querycontext

samsinsane commented on README.md:
```diff
@@ -1,42 +1,42 @@
+### Install Euclideon udSDK
+Euclideon udSDK can be obtained from [here](https://www.euclideon.com/vaultsdk/)
 
 Set the system variable for `VAULTSDK_HOME` on your operating system to the folder that vault SDK was downloaded to
```
```suggestion
Set the system variable for `UDSDK_HOME` on your operating system to the folder that udSDK was downloaded to
```

samsinsane commented on README.md:
```diff
@@ -1,42 +1,42 @@
-
-`setx VAULTSDK_HOME "[path to vault SDK]"`
+The following can also be achieved using the environment variables dialog.
+`setx UDSDK_HOME "[path to vault SDK]"`
```
```suggestion
`setx UDSDK_HOME "[path to udSDK]"`
```

samsinsane commented on README.md:
```diff
@@ -1,42 +1,42 @@
 e.g.
 
-`setx VAULTSDK_HOME "C:\Euclideon_vdk0.6.0"`
+`setx UDSDK_HOME "C:\Euclideon_udvdk1.0.0"`
```
```suggestion
`setx UDSDK_HOME "C:\Euclideon_udsdk1.0.0"`
```

samsinsane commented on README.md:
```diff
@@ -1,42 +1,42 @@
 Add the following command to your `.bashrc`
 
-`export VAUTLSDK_HOME="[path to vault sdk]"
+`export UDSDK_HOME="[path to vault sdk]"`
```
```suggestion
`export UDSDK_HOME="[path to udSDK]"`
```

samsinsane commented on README.md:
```diff
@@ -59,15 +59,15 @@ All packages are available through the pip repository and can be installed via
-This is the low level wrapper for Vault SDK to python. It implements a subset of the available C library as Python classes. Not all functionality
-of Vault SDK is currently implemented in this file. See the Vault SDK documentation for a full list of exposed functions.
+This is the low level wrapper for udSDK to python. It implements a subset of the available C library as Python classes. Not all functionality
+of udSDK is currently implemented in this file. See the Vault SDK documentation for a full list of exposed functions.
```
```suggestion
of udSDK is currently implemented in this file. See the udSDK documentation for a full list of exposed functions.
```

samsinsane commented on README.md:
```diff
@@ -59,15 +59,15 @@ All packages are available through the pip repository and can be installed via
 
 #### main.py
 
 This file contains the basic usage of vault SDK for rendering a UDS and writing it to file using Pillow.
```
```suggestion
This file contains the basic usage of udSDK for rendering a UDS and writing it to file using Pillow.
```

samsinsane commented on README.md:
```diff
@@ -59,15 +59,15 @@ All packages are available through the pip repository and can be installed via
+It demonstrates low level usage of functions and is a good starting point for users wanting to fully understand the process of generating a render using udSDK
 
-Usage: `python3 main.py yourvaultusername yourvaultpassword [pathtoUDS]` will run the sample code and write an image using the Vault Renderer
+Usage: `python3 main.py yourudStreamusername yourudStreampassword [pathtoUDS]` will run the sample code and write an image using the Vault Renderer
```
Really should look into using Python to prompt for this information instead of getting it saved on the command line.

samsinsane commented on README.md:
```diff
@@ -76,25 +76,27 @@ to a set of images. It is intended as a demonstration of generating renderings f
-for Vault SDK users wanting to develop their own client implementations and is designed to be easily extensible using python. It relies on numpy
+This example includes and example client written in python using the above wrappers for udSDKs libraries. It is intended as a starting point
+for udSDK users wanting to develop their own client implementations and is designed to be easily extensible using python. It relies on numpy
 and pyglet libraries in addition to Vault and Pillow
```
```suggestion
and pyglet libraries in addition to udSDK and Pillow
```

samsinsane commented on README.md:
```diff
@@ -76,25 +76,27 @@ to a set of images. It is intended as a demonstration of generating renderings f
 `App` represents the window context that the application runs in. It handles all user input and manages dispaches the draw commands to the
 various UI objects.
-`VDKViewPort` and it's derived objects represents an OpenGL quad which is to be textured using the frame buffer written to by each call of vdkRenderContext.Render
+`VDKViewPort` and it's derived objects represents an OpenGL quad which is to be textured using the frame buffer written to by each call of udRenderContext.Render
```
`VDKViewPort` should become `udSDKViewPort`.

samsinsane commented on camera.py:
```diff
@@ -1,12 +1,10 @@
 import numpy as np
+import pyglet
+
 import vault
```
Should change to `udSDK`

samsinsane commented on vault.py:
```diff
@@ -7,52 +7,54 @@
+    udSDK = CDLL(SDKPath)
   except OSError:
     logger.info(
       "No local Vault shared object/dll found in current working directory, trying path in VAULTSDK_HOME environment variable...")
```
```suggestion
      "No local udSDK shared object/dll found in current working directory, trying path in UDSDK_HOME environment variable...")
```

samsinsane commented on vault.py:
```diff
@@ -7,52 +7,54 @@
+    SDKPath = os.environ.get("UDSDK_HOME")
     if SDKPath == None:
-      raise FileNotFoundError("Environment variable VAULTSDK_HOME not set, please refer to Vault SDK documentation")
+      raise FileNotFoundError("Environment variable UDSDK_HOME not set, please refer to Vault SDK documentation")
```
```suggestion
      raise FileNotFoundError("Environment variable UDSDK_HOME not set, please refer to udSDK documentation")
```

samsinsane commented on vault.py:
```diff
@@ -7,52 +7,54 @@
     else:
       logger.error("Platform {} not supported by this sample".format(platform.system()))
       exit()
     logger.info("Using Vault SDK shared object located at {}".format(SDKPath))
```
```suggestion
    logger.info("Using udSDK shared object located at {}".format(SDKPath))
```

samsinsane commented on vault.py:
```diff
@@ -503,74 +707,176 @@ class vdkPointBufferF64(Structure):
+
+class udStreamer(Structure):
+  _fields_ = [
+    ("active", c_bool),
```
This isn't a bool, we explicitly removed `bool` from our API - if you come across any please report them.

samsinsane reviewed (changes requested):
There's a considerable number of references to Vault and VDK in these files.

Commit:
removed outdated vault references

modified logic for determining viewport sizing for easier manual placement

minor fixes to basic render and easy renderer;

samsinsane commented on README.md:
```diff
@@ -57,44 +57,46 @@ All packages are available through the pip repository and can be installed via
 
 UDS files can then be loaded by dragging and dropping from your OS shell (Explorer on Windows) into the window that is created.
 
 ![screenshot](./doc/clientScreenshot.png)
```
This screenshot contains `Vault` in the title bar of the Python program.

samsinsane reviewed (changes requested):
- `basicRender.py` contains numerous references to both `vault` and `vdk`
- `camera.py` contains reference to `vault`, several references to `vdk`
- `converter.py` contains numerous references to both `vault` and `vdk`
- `easyrenderer.py` contains numerous references to `vault`, `vdk` and `earth.vault.euclideon.com`
- `udSDK.py` contains a number of references to `vault` and `vdk`
- `viewport_experimental.py` contains a references to `vdk`

samsinsane reviewed (changes requested):
4 of the 6 files I previously mentioned still contain `vdk`, `vault` or `earth.vault.euclideon.com`.

samsinsane commented:
Also, the new screenshot says "udStream" but this wasn't called "Vault Client" before - please use the correct branding. Additionally, you should limit your screenshots to the application itself, your taskbar provides no additional context here and only serves as a distraction from the content.

samsinsane commented on converter.py:
```diff
@@ -1,15 +1,15 @@
-Requires Vault SDK shared libraries (.so or .dll)
-Converts any point cloud format supported by Vault SDK to UDS format 
+General purpose command line converter for Euclideon udSDK using python
+Requires udSDK SDK shared libraries (.so or .dll)
```
```suggestion
Requires udSDK shared libraries (.so or .dll)
```

samsinsane commented on converter.py:
```diff
@@ -37,43 +37,41 @@
+def login():
     """
-    Connect to vault server and requests a license
+    Connect to udStream server and requests a license
```
```suggestion
    Connect to the udStream server
```

samsinsane reviewed (changes requested):
The latest iteration of the screenshot doesn't have the same window title as what is set in the code.

Once all of the issues are resolved, the commits should be squished to remove the multiple iterations of the screenshot and the uninformative commit messages. Additionally, this PR should really only contain changes related to updating to udSDK, keep your changes isolated so that it's easier to see where things went wrong.
